test/test_client.lua

```diff
@@ -0,0 +1,88 @@
+local nakama = require "nakama.nakama"
+local test_engine = require "nakama.engine.test"
+
+
+context("Nakama client", function()
+
+	before(function()
+		test_engine.reset()
+	end)
+	after(function() end)
+
+	local function config()
+		return {
+			host = "127.0.0.1",
+			port = 7350,
+			use_ssl = false,
+			username = "defaultkey",
+			password = "",
+			engine = test_engine,
+			timeout = 10, -- connection timeout in seconds
+		}
+	end
+
+	local function pprint(t)
+		for k,v in pairs(t) do
+			print(k, v)
+		end
+	end
+
+	test("It should be able to create a client", function()
+		local config = config()
+		local client = nakama.create_client(config)
+		assert_not_nil(client)
+	end)
+
+	test("It should be able to call REST API functions", function()
+		local token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOiI1MjJkMGI5MS00NmQzLTRjY2ItYmIwYS0wNTFjYjUyOGNhMDMiLCJ1c24iOiJicml0emwiLCJleHAiOjE2NjE1OTA5Nzl9.r3h4QraXsXl-XmGQueYecjeb6223vtd1s-Ak1K_FrGM"
+		local refresh_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOiI1MjJkMGI5MS00NmQzLTRjY2ItYmIwYS0wNTFjYjUyOGNhMDMiLCJ1c24iOiJicml0emwiLCJleHAiOjE2NjE1ODczNzl9.AWASctuZx9A8YliCLSj9jtOi4fuXUZaWtRdNz1mMEEw"
+		local data = {
+			token = token,
+		}
+		local url_path = "/v2/account/authenticate/email"
+		test_engine.set_http_response(url_path, data)
+
+		coroutine.wrap(function()
```

Comment:
This line intermittently throws an error in the test.

> attempt to call a nil value (global: 'unpack')

Reply:
It does? Is this when you test locally? On which OS? And with which Lua version?

Reply:
Oh, I'm getting this as well. It's a Lua version issue. I've fixed it by using a fork of the unit test framework that supports both `unpack` and `table.unpack`